src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs

```diff
@@ -1019,6 +1019,20 @@ public void DefaultExecuteCmdlet()
                         }
                     }
                 }
+                else if (this.VM.StorageProfile?.ImageReference?.SharedGalleryImageId != null) {
+                    // do nothing, send message to use TrustedLaunch.
+                    if (this.AsJobPresent() == false) // to avoid a failure when it is a job. Seems to fail when it is a job.
```

Comment:
I've run manual tests on these changes and it works, doesn't break any scenarios.